// API for DBOS components
// Interface between client and scheduler

syntax = "proto3";

package dbos_scheduler;

import "dbos_status.proto";

// TODO: need to discuss the interface design; subject to changes.
service Frontend {
  // Client sends a task to the scheduler.
  rpc SubmitTask(SubmitTaskRequest) returns (SubmitTaskResponse) {}

  // Heartbeat to scheduler.
  rpc Heartbeat(HeartbeatRequest) returns (HeartbeatResponse) {}
}

// Message types for Frontend
message SubmitTaskRequest {
  int64 requirement = 1; // resource requirement.
  int64 execTime = 2;    // execution time in nanoseconds.
}

message SubmitTaskResponse {
  DbosStatusEnum status = 1;
}

message HeartbeatRequest {
  string msg = 1;
}

message HeartbeatResponse {
	DbosStatusEnum status = 1;
}

